### PR TITLE
10959-Only send impression events from background service and make thread safe

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
https://swrvedev.jira.com/browse/SWRVE-10959

@shanetmoore Further changes to my original PR. These changes allow new events to be stored and only send those new events. 

Also made it thread safe.

### CODE REVIEW

File: SwrveSDK/src/main/java/com/swrve/sdk/SendEventsManager.java [hunk 1]
```diff
@@ -61,13 +106,13 @@ public void onResponse(boolean shouldDelete) {
...
             } catch (JSONException je) {
-                SwrveLogger.e(LOG_TAG, "Unable to generate event batch", je);
+                SwrveLogger.e(LOG_TAG, "Unable to generate event batch, and send events", je);
             }
         }
         return eventsSent;
     }
 
-    protected void postBatchRequest(String postData, final IPostBatchRequestListener listener) {
+    private void postBatchRequest(final String postData, final IPostBatchRequestListener listener) {
```
Comment (shanethmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SendEventsManager.java]: Does it need to be final to be stored?

**
NVM I see the onException log

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java [hunk 2]
```diff
@@ -3,19 +3,20 @@
...
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
 
 public class SwrveWakefulService extends IntentService {
 
     private static final String LOG_TAG = "SwrveWakeful";
     public static final String EXTRA_SEND_EVENTS = "swrve_wakeful_events";
 
+    private Swrve swrve = (Swrve) SwrveSDKBase.getInstance();
```
Comment (shanethmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java:18]: How will this behave if they haven't initialized the sdk?

Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java:18]: If they haven't initialized the sdk, then SwrveWakefulService will  not have been called. This is only being used by the Locations sdk (so far).

**Comment by shanethmoore** [comment]
This all looks good. Is this so only Location events are sending or does this stop us sending events when the user restarts the app (causing a session start)

Only other suggestion I'd have is changing the name of `public class SendEventsManager {` to SwrveEventsManager. This is just to keep it inline with the rest of our classes and also to keep logic or implementation details out of the name.  Totally up to yourself though.

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java (hunk 2 above)
Comment (shanethmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java:18]: ah, ok noted

**Comment by dominicmarmionswrve** [comment]
Yes, this is so only the location events, or whatever is passed to storeAndSendEvents is actually sent. If these don't get sent in background (when no network), they'll get sent next time the app is started in the normal way.

Cool, will rename to SwrveEventsManager.

**Pull request merged**